dockers/docker-fpm-frr/bgpcfgd

```diff
@@ -1,124 +1,219 @@
+        if not value:
+            return False
+        if isinstance(value, netaddr.IPNetwork):
+            addr = value
+        else:
+            try:
+                addr = netaddr.IPNetwork(str(value))
+            except:
+                return False
+        return addr.version == 6
 
 
 class BGPConfigManager(object):
     def __init__(self, daemon):
-        self.daemon = daemon
         self.bgp_asn = None
-        self.bgp_message = Queue.Queue(0)
+        self.meta = None
+        self.bgp_messages = []
+        self.peers = self.load_peers() # we can have bgp monitors peers here. it could be fixed by adding support for it here
+        fabric = TemplateFabric()
+        self.bgp_peer_add_template = fabric.from_file('bgpd.peer.conf.j2')
+        self.bgp_peer_del_template = fabric.from_string('no neighbor {{ neighbor_addr }}')
         daemon.add_manager(swsscommon.CONFIG_DB, swsscommon.CFG_DEVICE_METADATA_TABLE_NAME, self.__metadata_handler)
         daemon.add_manager(swsscommon.CONFIG_DB, swsscommon.CFG_BGP_NEIGHBOR_TABLE_NAME, self.__bgp_handler)
 
+    def load_peers(self):
+        peers = set()
+        command = ["vtysh", "-c",  "show bgp neighbors json"]
+        rc, out, err = run_command(command)
+        if rc == 0:
+            js_bgp = json.loads(out)
+            peers = set(js_bgp.keys())
+        return peers
+
     def __metadata_handler(self, key, op, data):
         if key != "localhost" \
             or "bgp_asn" not in data \
                 or self.bgp_asn == data["bgp_asn"]:
             return
 
-        # TODO add ASN update commands 
+        # TODO add ASN update commands
 
+        self.meta = { 'localhost': data }
         self.bgp_asn = data["bgp_asn"]
         self.__update_bgp()
 
     def __update_bgp(self):
-        while not self.bgp_message.empty():
-            key, op, data = self.bgp_message.get()
-            syslog.syslog(syslog.LOG_INFO, 'value for {} changed to {}'.format(key, data))
+        cmds = []
+        for key, op, data in self.bgp_messages:
+            add_peer_chunk = self.bgp_peer_add_template.render(DEVICE_METADATA=self.meta, neighbor_addr=key, bgp_session=data)
+            del_peer_chunk = self.bgp_peer_del_template.render(neighbor_addr=key)
             if op == swsscommon.SET_COMMAND:
-                command = "vtysh -c 'configure terminal' -c 'router bgp {}' -c 'neighbor {} remote-as {}'".format(self.bgp_asn, key, data['asn'])
-                run_command(command)
-                if "name" in data:
-                    command = "vtysh -c 'configure terminal' -c 'router bgp {}' -c 'neighbor {} description {}'".format(self.bgp_asn, key, data['name'])
-                    run_command(command)
-                if "admin_status" in data:
-                    command_mod = "no " if data["admin_status"] == "up" else ""
-                    command = "vtysh -c 'configure terminal' -c 'router bgp {}' -c '{}neighbor {} shutdown'".format(self.bgp_asn, command_mod, key)
```

Comment:
how do we control the bgp sesssion up/down command now?

Reply:
Currently by removing whole other session and applying the new one.
Currently I support only add/remove. We need to introduce modify to support correct shutdown/no shutdown functionality. Or make a special case for shutdown/no shutdown

Reply:
shur / unshut is hard requirement.